File: String/Longest Common Anangram/LongestCommonAnagram.py
# Python 3 implementation to find
# the length of the longest common
# anagram subsequence

SIZE = 26

# function to find the length of the
# longest common anagram subsequence
def longCommonAnagramSubseq(str1, str2,
								n1, n2):

	# List for storing frequencies
	# of each character
	freq1 = [0] * SIZE
	freq2 = [0] * SIZE
	
	l = 0
	
	# calculate frequency of each
	# character of 'str1[]'
	for i in range(n1):
		freq1[ord(str1[i]) -
			ord('a')] += 1
	
	# calculate frequency of each
	# character of 'str2[]'
	for i in range(n2) :
		freq2[ord(str2[i]) -
			ord('a')] += 1
	
	# for each character add its
	# minimum frequency out of
	# the two strings in 'len'
	for i in range(SIZE):
		l += min(freq1[i], freq2[i])
	
	# required length
	return l							

# Driver Code
if __name__ == "__main__":
	
	str1 = "abdacp"
	str2 = "ckamb"
	n1 = len(str1)
	n2 = len(str2)
	print("Length = ",
		longCommonAnagramSubseq(str1, str2,
									n1, n2))
 

File: String/Longest Common Anangram/LongestCommonAnagram.cpp
// Java implementation to find
// the length() of the longest
// common anagram subsequence
import java.io.*;

class GFG
{
	static int SIZE = 26;
	
	// function to find the
	// length() of the longest
	// common anagram subsequence
	static int longCommonAnagramSubseq(String str1,
									String str2,
									int n1, int n2)
	{
		// hash tables for
		// storing frequencies
		// of each character
		int []freq1 = new int[SIZE];
		int []freq2 = new int[SIZE];
		
		for(int i = 0; i < SIZE; i++)
		{
			freq1[i] = 0;
			freq2[i] = 0;
		}
		
		int len = 0;
		
		// calculate frequency
		// of each character of
		// 'str1[]'
		for (int i = 0; i < n1; i++)
			freq1[(int)str1.charAt(i) - (int)'a']++;
		
		// calculate frequency
		// of each character
		// of 'str2[]'
		for (int i = 0; i < n2; i++)
			freq2[(int)str2.charAt(i) - (int)'a']++;
		
		// for each character add
		// its minimum frequency
		// out of the two Strings
		// in 'len'
		for (int i = 0; i < SIZE; i++)
			len += Math.min(freq1[i],
							freq2[i]);
		
		// required length()
		return len;
	}							
	
	// Driver Code
	public static void main(String args[])
	{
		String str1 = "abdacp";
		String str2 = "ckamb";
		int n1 = str1.length();
		int n2 = str2.length();
		System.out.print("Length = " +
				longCommonAnagramSubseq(str1, str2,
										n1, n2));
	}
}


